pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -199,6 +200,15 @@ func (c *Cluster) checkHealth() error {
 			continue
 		}
 
+		//retry and don't failover if mon POD is not scheduled.
+		isScheduled, err := isMonScheduled(c.context.Clientset, c.Namespace, mon.Name)
```

Comment:
I don't see how this handles the second scenario in [this comment](https://github.com/rook/rook/issues/7776#issuecomment-830137688). If a mon is not portable, we show go ahead and allow the failover to attempt to start another mon. If another node is available, the failover could proceed successfully.

Reply:
I've added that now. Though I'm not sure what should be correct way to determine if the mon POD is portable or not.  
Is `c.spec.Mon.VolumeClaimTemplate != nil` check enough to ensure that mon POD is portable ?

Reply:
We don't really know if the mon is portable since the storage class backing the PVC might or might not be portable. I don't think we need to treat this specially though. To be safe, we can just allow the failover to be triggered.